### DISCUSSION TITLE
Add multiple active field indicators to one html element

### DISCUSSION THREAD

**Discussion by Perni1984** [tinacms/tinacms#2604] [Help]
We would have the need to add multiple active field indicators for different fields to one html element of our component as shown here.

As `data-tinafield` can only hold one value we would need to decide which one makes the most sense, but would loose the ability for the other fields.

![image](https://user-images.githubusercontent.com/9039956/154836741-6c8e0694-db49-435c-b6dc-771c2aecf1f7.png)

Do you have any idea of how we can accomplish multiple active field indicators on one html element - without cluttering too much our frontend component html structure?

Maybe have something like data-tinafield, `data-tinafield1`, `data-tinafield2` ... `data-tinafield*` ?

**Comment by jeffsee55** [comment]
@Perni1984 is it reasonable to put those fields into an `object`? At that point you'll be able to just reference the name of that object

```ts
{
  type: 'object',
  name: 'imageAndSuch', // put this field in `data-tinaField`
  fields: [
    {
      type: "image",
      name: "imageSrc",
      ...
    },
    {
      type: "boolean",
      name: "aboveTheFold",
      ...
    }
  ]
}
```

**Reply by Perni1984** [reply]
@jeffsee55 That could be a "workaround" for this particular use case, but IMHO it would not be a scalable solution - because at least in our case the schema mimics the typescript API of the underlying frontend component - and I wouldn't want to change the frontend component API just because of the active field indicator.

I have not looked into the code, but don't you use some kind of css selector to query the component you put the active field indicator around? Couldn't this css selector be a bit more widened, e.g. to not query for exact matches, but for any occurence?

Because then I could put in the `data-tinafield` attribute a list of fieldnames that should match (e.g. comma separated maybe?)

**Reply by jeffsee55** [reply]
@Perni1984 we're hoping to provide the ability to click on the active field indicator (somehow) and link _back_ to the form, if the indicator is linking to 2 fields, what would you expect the link-back behavior to be?

**Reply by Perni1984** [reply]
@jeffsee55 I think I would expect to see two links (2 icons?!?) on the element, each one linking to a specific field. But to be honest I didn't think about this too much yet. Another approach we could do already now is to add `<span></span>`'s around the element, just for the sake of being able to put data-tinaFields somewhere. Maybe we're also going into this direction, the thing I don't like about it is that cms-editing concerns are coming into our frontend components.
